File: exercise/c#/Day12/Day12/Greeter.cs
namespace Day12
{
    public class Greeter
    {
        public string? Formality { get; set; }

        public string Greet()
        {
            if (Formality == null)
            {
                return "Hello.";
            }

            if (Formality == "formal")
            {
                return "Good evening, sir.";
            }
            else if (Formality == "casual")
            {
                return "Sup bro?";
            }
            else if (Formality == "intimate")
            {
                return "Hello Darling!";
            }
            else
            {
                return "Hello.";
            }
        }
    }
}

File: exercise/ts/day12/src/greeter.ts
export class Greeter {
    private formality: string | null = null;

    greet(): string {
        if (this.formality == null) {
            return 'Hello.';
        }

        if (this.formality === 'formal') {
            return 'Good evening, sir.';
        } else if (this.formality === 'casual') {
            return 'Sup bro?';
        } else if (this.formality === 'intimate') {
            return 'Hello Darling!';
        } else {
            return 'Hello.';
        }
    }

    setFormality(formality: string): void {
        this.formality = formality;
    }
}